test/components/structures/RoomView-test.js

```diff
@@ -0,0 +1,71 @@
+var React = require('react');
+var expect = require('expect');
+var sinon = require('sinon');
+var ReactDOM = require("react-dom");
+
+var sdk = require('matrix-react-sdk');
+var RoomView = sdk.getComponent('structures.RoomView');
+var peg = require('../../../src/MatrixClientPeg');
+
+var test_utils = require('../../test-utils');
+var q = require('q');
+
+var Skinner = require("../../../src/Skinner");
+var stubComponent = require('../../components/stub-component.js');
+
+describe('RoomView', function () {
+    var sandbox;
+
+    beforeEach(function() {
+        sandbox = test_utils.stubClient();
+
+        this.oldTimelinePanel = Skinner.getComponent('structures.TimelinePanel');
+        this.oldRoomHeader = Skinner.getComponent('views.rooms.RoomHeader');
+        Skinner.addComponent('structures.TimelinePanel', stubComponent());
+        Skinner.addComponent('views.rooms.RoomHeader', stubComponent());
+
+        peg.get().credentials = { userId: "@test:example.com" };
+    });
+
+    afterEach(function() {
+        sandbox.restore();
+        Skinner.addComponent('structures.TimelinePanel', this.oldTimelinePanel);
```

Comment:
it would be nice to find a way to do this automatically via the sandbox... this will do for now though.